scripts/data/common/killed-in-gaza/extract_20240501.py

```diff
@@ -0,0 +1,87 @@
+import re
+import logging
+import pandas as pd
+
+# set up logging
+logging.basicConfig(
+    level=logging.DEBUG,
+    format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+    datefmt='%Y-%m-%d %H:%M:%S',
+    handlers=[
+        logging.StreamHandler()
+    ]
+)
+logging.getLogger(__name__)
+
+logging.info("loading input xlsx file")
+working_dir = "scripts/data/common/killed-in-gaza/"
+# set file path and header and load the file
+# skipping first row which is file header
+file_path = f"{working_dir}raw/30-04_1_raw.xlsx"
+header_en = ["sex", "location", "age", "id", "name_ar_raw", "serial_no"]
+df = pd.read_excel(file_path, skiprows=1, names=header_en)
+
+logging.info("applying file specfic cleaning on the file")
+# removing one column which has empty values for all columns except sex
+df = df[df['sex'] != 'ل']
+
+# replace empty strings (if exist) in id and name_ar_raw
+columns_to_replace = ['id', 'name_ar_raw']
+df[columns_to_replace] = df[columns_to_replace].replace('', pd.NA)
+rows_with_nan_name_ar_raw = df[df['name_ar_raw'].isna()]
+
+# for 6 rows where name is empty and id is not empty
+# id column incorrectly contains names
+mask = df['name_ar_raw'].isna() & ~df['id'].isna()
+df.loc[mask, 'name_ar_raw'] = df.loc[mask, 'id']
+df.loc[mask, 'id'] = pd.NA
+
+# fixing id column
+# - replace zero integer with string
+# - for given incorrect ids replace them with NA -> 000000000 -> missing-20240501-<serial_no>
+# - remove duplicates based on id 
+df.loc[df['id'] == 0, 'id'] = '0'
+incorrect_ids = [
+    'طفلةة','تحت الركام','-','مولود','_','طفلة','.','ال اعرف','ك',
+    'طفلل','طففل','طفل','طفلة حديث','طفل...خدج','طفل ...خدج','طفللل', '0'
+]
+df['id'] = df['id'].replace(incorrect_ids, pd.NA)
+df['id'] = df['id'].fillna('000000000')
+df.loc[df['id'] == '000000000', 'id'] = 'missing-20240501-' + df['serial_no'].astype(int).astype(str)
+logging.info("following dupes are removed")
+# logging.info([df.duplicated(['id'], keep=False)])
+df = df.drop_duplicates(subset='id')
+
+# calculate dob from age keeping the date the list was published
+# as the reference date
+df['age'] = pd.to_numeric(df['age'], downcast='integer', errors='coerce')
+df['age'] = df['age'].astype('Int64')
+reference_date = pd.to_datetime('2024-04-30')
+df['dob'] = reference_date - pd.to_timedelta(df['age']*365, unit='D')
```

Comment:
should we only be doing this if the age & dob are inconsistent OR if one of the two isn't provided? seems like this is overwriting the values in the sheet regardless?

the reference date approach doesn't work i think since a bunch of names in the list were from prior list releases which had different reference dates

Reply:
Ah sorry, I forgot they removed that column as well....

I think we should probably move your logic here

<img width="590" alt="Screenshot 2024-06-22 at 2 40 53 PM" src="https://github.com/TechForPalestine/palestine-datasets/assets/1047502/39a36f63-8f8e-41b8-8ecf-2bfae0d03654">

from what I can tell, if the DOB is precalculated before the reconcile script, it'll overwrite the existing one unless it's invalid. And since you're calculating it based on the age it's unlikely to be considered invalid.

I can help move your logic over if you're not comfortable with typescript, let me know!

Reply:
(what I mean is, if you didn't have the precalculated DOB in your initial python script, the new records would all hit this condition and be skipped / not added to the merged file, but we can add your DOB calculation here if the DOB is not in both and the age is different in the new record if that makes sense - i think in that case we wouldn't be doing it too often for the old records and all the time for the new ones)

Reply:
Thanks for sharing the output 🙏🏻  The lower change numbers look better 😌 

I think with those small changes to the `addExistingAge` function we're good to go!